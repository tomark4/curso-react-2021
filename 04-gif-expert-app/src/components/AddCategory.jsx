import {useState} from 'react'
import PropTypes from 'prop-types';

const AddCategory = ({onAdd}) => {

    const [value, setValue] = useState('')

    const handleSubmit = (e) => {
        e.preventDefault();
        if(value.trim().length > 2){
            onAdd(value);
            setValue("");
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input
            type="text"
            name="value"
            value={value}
            onChange={(e) => setValue(e.target.value) }/>
        </form>
    )
}

AddCategory.propTypes = {
    onAdd: PropTypes.func.isRequired
};

export default AddCategory
