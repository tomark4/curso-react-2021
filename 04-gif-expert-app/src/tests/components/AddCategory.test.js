import {shallow} from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Pruebas en component AddCategory', () => {
  const onAdd = () => {}
  const wrapper = shallow( <AddCategory onAdd={onAdd}/>);

  test('should show component correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de cambiar la caja de texto', () => {
    const input = wrapper.find('input[type="text"]');
    const value = 'hola mundo'
    input.simulate('change',{ target: { value }});

    const inputAfter = wrapper.find("input[type='text']");
    expect(inputAfter.prop("value")).toBe(value);
  })


});

