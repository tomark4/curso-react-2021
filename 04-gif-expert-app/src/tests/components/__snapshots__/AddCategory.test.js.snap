// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en component AddCategory should show component correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="value"
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
